### PR TITLE
BDOG-1540

### PR DESCRIPTION
**Pull request by tomwadeson**
Various TLC changes to `init-service`, along with an example of an HTTP-layer integration test for backend and frontend projects.

### CODE REVIEW

File: scripts/test/integration-tests.py [hunk 1]
```diff
@@ -88,8 +94,8 @@ def test_created_code_compiles(self):
         self.runCreate(project_prefix + '-frontend-mongo', 'BACKEND', with_mongo=True)
         self.runCreate(project_prefix + '-library', 'LIBRARY')
 
-        self.runSbtCommand(projects, 'compile')
```
Comment (tomwadeson) [inline scripts/test/integration-tests.py:91]: Given that `compile` is implied by `test`, these tests run a bit quicker by not having to spin up the SBT server twice

File: templates/library/src/test/scala/uk/gov/hmrc/hello/HelloWorldSpec.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmrc.hello
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class HelloWorldSpecs extends AnyWordSpec with Matchers {
```
Comment (colin-lamed) [inline templates/library/src/test/scala/uk/gov/hmrc/hello/HelloWorldSpec.scala:6]: should rename the class to `HelloWorldSpec` too?

File: templates/service/it/HealthEndpointIntegrationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneServerPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.ws.WSClient
+
+class HealthEndpointIntegrationSpec
```
Comment (colin-lamed) [inline templates/service/it/HealthEndpointIntegrationSpec.scala:11]: Now we've added this, can we remove `MicroserviceIntegrationSpec`?
MicroserviceIntegrationSpec isn't really testing `AppConfig` (which could be tested without inflating the whole application) - it's there to serve the same purpose as `HealthEndpointIntegrationSpec` - i.e. does the app start as a whole.

**Review by colin-lamed — APPROVED** [review]
Some additional ideas of things we could look at. Appreciate some require investigation, and so most likely out of scope for this change.:

- scala version is a variable, but silencerVersion is fixed - they are related, so should be both variables in the same place? They can also be updated to scala 2.12.15, silencerVersion 1.7.7
- Clean up logback.xml - I think it's full of things we don't need: `STDOUT_IGNORE_NETTY`, `CONNECTOR_LOG_FILE`. Be interesting to see if they are used by anyone (I think the netty one is obsolete)
- application.conf - should metrics and auditing be disabled by default? They are enabled in the environments - are they required for development? (It could make the tests easier to run, without having to override the values).

**Review by colin-lamed — DISMISSED** [review]

**Review by colin-lamed — APPROVED** [review]

**Pull request merged**